.github/workflows/binaries.yml

```diff
@@ -0,0 +1,61 @@
+name: Generate binaries
+on:
+  release:
+    types:
+      - created
+jobs:
+  binaries:
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            asset_name: api-linux-amd64.tar.gz
+            compress_cmd: tar -czvf
+            build_cmd: docker run -v $PWD:/data golang:1.19 bash -c "cd /data && go build ./cmd/api"
+          - os: windows-latest
+            asset_name: api-windows-amd64.zip
+            compress_cmd: tar.exe -a -c -f
+            build_cmd: go build -o api ./cmd/api
+          - os: macos-latest
+            asset_name: api-darwin-amd64.tar.gz
+            compress_cmd: tar -czvf
+            build_cmd: go build -o api ./cmd/api
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: v1.19.x
+      - name: Build binary
+        run: ${{ matrix.build_cmd }}
+      - name: Pack output
+        run: ${{ matrix.compress_cmd }} ${{ matrix.asset_name }} api
+      - name: Upload binary
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: ${{ matrix.asset_name }}
+          asset_name: ${{ matrix.asset_name }}
+          tag: ${{ github.ref_name }}
+          overwrite: true
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          file: ./cmd/api/Dockerfile
+          push: true
+          tags: textile/tableland:latest,textile/tableland:${{ github.ref_name }}
+          platforms: linux/amd64, linux/arm64
```

Comment:
This line is a bit magical because `buildx` will leverage that QEMU is installed in the VM, and to build for architectures that aren't the native one in the VM, it will run all the Docker process using QEMU.

This means that the `Dockerfile` image building for `linux/arm64` will be run in QEMU (since we only have `amd64` machines in GHA). It takes 10x the time compared to `linux/amd64` which is run natively, but still 10mins is a reasonable time to wait.

Note that I tried the same idea for binaries, so I can run a `golang:1.19` in a simulated `arm64` machine. The problem was that I needed the GCC XCode compiler installed in Ubuntu to do the CGO cross compiling, which was nasty.
In Docker, M1 machines could run images that match for their architecture... that's why these docker images should be runnable in M1.